p2p/service.py

```diff
@@ -94,9 +103,27 @@ def __init__(self, token: CancelToken=None) -> None:
             if finished_callback is not None:
                 finished_callback(self)
 
+    def run_child_service(self, child_service: 'BaseService') -> 'asyncio.Future[Any]':
+        """
+        Run a child service and keep a reference to it to be considered during the cleanup.
+        """
+        self._child_services.append(child_service)
+        return asyncio.ensure_future(child_service.run())
+
     async def cleanup(self) -> None:
-        """Run the service's _cleanup() coroutine and sets the cleaned_up event."""
-        await self._cleanup()
+        """
+        Run the ``_cleanup()`` coroutine and set the ``cleaned_up`` event after the service as
+        well as all child services finished their cleanup.
+
+        The ``_cleanup()`` coroutine is invoked before the child services may have finished
+        their cleanup.
+        """
+
+        await asyncio.gather(*[
+            svc.cleaned_up.wait()
```

Comment:
Thoughts on just `svc -> child_service`?  (I have grown to dislike most abbreviations)

Reply:
Fine by me!